src/components/Camera.tsx

```diff
@@ -0,0 +1,62 @@
+import { useRef, useEffect, useState } from "react";
+import "./Camera.css";
+
+export const Camera = () => {
+  const [error, setError] = useState<string | null>(null);
+  const [isEnabled, setIsEnabled] = useState(false);
+  const videoRef = useRef<HTMLVideoElement | null>(null);
+
+  useEffect(() => {
+    if (!isEnabled || !videoRef.current) return;
+
+    let stream: MediaStream | null = null;
+
+    const startStream = async () => {
+      try {
+        stream = await navigator.mediaDevices.getUserMedia({
```

Comment:
`startStream` starts twice if enabled is true on page load:
```
  const [isEnabled, setIsEnabled] = useState(true);
```

One of the streams started (for which cleanup is called) should be correctly terminated:
stream.getTracks().forEach((track) => track.stop());
and should not update srcObject